gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/NotificationController.groovy

```diff
@@ -48,4 +51,21 @@ class NotificationController {
     notificationService.getNotificationConfigs(type, application)
   }
 
+  /**
+   * Provides a generic callback API for notification services to call, primarily in response to interactive user
+   * action (e.g. clicking a button in a message).
+   *
+   * @param source The unique ID of the calling notification service (e.g. "slack")
+   * @param request The callback request
+   * @return
+   */
+  @RequestMapping(
```

Comment:
Spring recommends using the `@{method}Mapping` annotations over the generic `@RequestMapping` annot. `s/Request/Post`